modules/auxiliary/scanner/scada/modbusclient.rb

```diff
@@ -369,6 +386,31 @@ def write_registers
       print_error("Unknown answer")
     end
   end
+  
+  def read_id
+    @function_code = 0x2b
+    print_status("Sending READ ID...")
+    response = send_frame(make_read_id_payload)
+    if response.nil?
+      print_error("No answer for the READ REGISTER")
+    elsif response.unpack("C*")[7] == (0x80 | @function_code)
+      handle_error(response)
+    elsif response.unpack("C*")[7] == @function_code
+      @ObjCnt = response[13].unpack("C")[0]
+      @ObjIdPos=14
+      begin
+    @ObjLen = response[@ObjIdPos+1].unpack("C")[0]
```

Comment:
Questioning why `ObjLen` needs to be an instance variable, looks like this could just be scoped to the local function instead.

Reply:
You're right. It's useless here